src/libsyntax/ast.rs

```diff
@@ -220,8 +222,9 @@ impl Path {
     }
 }
 
-/// A segment of a path: an identifier, an optional lifetime, and a set of
-/// types.
+/// A segment of a path: an identifier, an optional lifetime, and a set of types,
```

Comment:
Since the first line is used not just in the documentation for the item itself but also in other contexts (e.g. when the item is referred to in other types), it should always be able to stand on its own. Basically, don't end it with a comma, just use a period instead and start the second paragraph with a capital letter :)